### DISCUSSION TITLE
Support Prometheus /graph api

### DISCUSSION THREAD

**Discussion by roidelapluie** [grafana/grafana#42316] [Observability]
**What would you like to be added**:

I would be interesting to see if Grafana can expose a Prometheus /graph-compatible endpoint. The endpoint would redirect to Grafana explore.

**Why is this needed**:

Prometheus can send a graph URL to alertmanager. It would be great if in the alertmanager notifications, we could apply a replaceAll template that switches from Prometheus /graph to grafana /prometheus/graph with e.g. a `replaceAll`. The user would then be redirected to /explore, with the correct query.

**Comment by seb-koch** [comment]
Loving the idea! I currently hacked it together right in the receiver configuration by using a set of `reReplaceAll` calls. Here is a working example where we replace the Prometheus Url with a Grafan Explore call including the Prometheus expression. The base URL has to be encoded as otherwise the link breaks when viewed in the browser (i.e. in Opsgenie).

```
note: '{{ range .Alerts }}{{ reReplaceAll ".*expr\\=" "https%3A%2F%2Fdomain.tld%2Fgrafana%2Fexplore%3ForgId%3D1%26left%3D%5B%5C%22now-15m%5C%22%2C%5C%22now%5C%22%2C%5C%22Prometheus%5C%22%2C%7B%5C%22exemplar%5C%22%3Atrue%2C%5C%22expr%5C%22%3A%5C%22" .GeneratorURL | reReplaceAll "&g0.tab=1" "%5C%22%7D%5D" }}{{end}}'
```

**Reply by aditya-opsverse** [reply]
Hello @seb-koch
I have tried the above solution that you gave. Unfortunately that didn't work. The URL is still broken.
Could you please help me in where I am going wrong?
Thanks

**Reply by seb-koch** [reply]
Hi, I don't have access to that system anymore unfortunately and thus can't reproduce it. Sorry, that I can't help here.

**Comment by ludov04** [comment]
Would love to have this

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82943. Feel free to continue the discussion around this there. Thank you!
